### PR TITLE
LOCALE: German translation

### PR DESCRIPTION
**Pull request by Justman100**

### CODE REVIEW

**Comment by Justman100** [comment]
@merefield

File: lib/bot_commands.rb [hunk 1]
```diff
@@ -91,7 +91,7 @@ def self.manage_discord_commands(bot)
                 associated_user = UserAssociatedAccount.find_by(provider_uid: instance[2..19], provider_name: 'discord')
                 if associated_user.nil?
                   discord_username = event.bot.user(instance[2..19]).username
-                  raw = raw.gsub(instance, I18n.t("discord_bot.commands.disccopy.mention_prefix", discord_username: discord_username) + instance[21..])
+                  raw = raw.gsub(instance, "discord_%{discord_username}", discord_username: discord_username + instance[21..])
```
Comment (merefield) [inline lib/bot_commands.rb]: Hey there, I'd prefer not to lose the flexibility of a locale string setting here.  It's good practice to make all exposed strings on the interface translatable. We also have a paying client using this plugin, so need to be careful not to disturb their potential overrides.

File: lib/bot_commands.rb [hunk 2]
```diff
@@ -115,7 +115,7 @@ def self.manage_discord_commands(bot)
             end
 
             if topic_index == 0 && destination_topic.nil?
-              raw = raw.blank? ?  I18n.t("discord_bot.commands.disccopy.discourse_topic_contents", channel: event.channel.name) : raw
+              raw = raw.blank? ?  "%{channel}", channel: event.channel.name : raw
```
Comment (merefield) [inline lib/bot_commands.rb]: Ditto, I'd prefer not to lose the flexibility of a locale string setting here.

**Comment by merefield** [comment]
Thanks for the updated German!  I've raised some comments.

**Review by merefield — CHANGES_REQUESTED** [review]

**Comment by merefield** [comment]
I've taken a quick look.  btw, it's good practice to get into the habit of naming the PR something obvious and relevant and providing a short summary in the OP (although latter probably not necessary in this instance.)

**Comment by Justman100** [comment]
> I've taken a quick look. btw, it's good practice to get into the habit of naming the PR something obvious and relevant and providing a short summary in the OP (although latter probably not necessary in this instance.)

One does not need to declare a stand-alone variable as a string! It is just a variable! They will not be translated! And I call all my PRs _Update_ on purpose, in case further commits should come, which would then differ from the title

**Comment by merefield** [comment]
> It is just a variable! They will not be translated!

It's not just about translation.  We can use the locale string to change the style of the username at runtime.  If we want something different to "discord_" we have the flexibility to change it.

In general locale strings are not just about translation.  They also provide an ability for admins to customise the appearance of their own specific install as they see fit.

In any case, this is in Production and being used by a business, so we can't change this.  Please remove these changes.

The German translations are welcome and appreciated.

**Comment by merefield** [comment]
> And I call all my PRs _Update_ on purpose, in case further commits should come, which would then differ from the title

So make your PR draft until you are ready for it to be reviewed and don't @ me until you are ready and able to provide a  PR which has a more useful (and relevant) name.  Of course it's an "Update" :) Let's have a more useful name please, like "German translation" - how hard is that really? (ok, I will do this myself: done!).

Oh and don't get so defensive that you start to defy rationality!  You've asked me to consider your PR  for merging so I'll review it :). Do not be surprised if ever some of your changes are not accepted as is as maintainers will often be opinionated and that's ok - it's their rodeo.  It's also an opportunity to learn (for both of us).

On the contrary, you should be grateful I looked at this during my weekend!

Are you trying to win a collaborator here or annoy someone? :)

**Comment by Justman100** [comment]
> > And I call all my PRs _Update_ on purpose, in case further commits should come, which would then differ from the title
> 
> So make your PR draft until you are ready for it to be reviewed and don't @ me until you are ready and able to provide a PR which has a more useful (and relevant) name. Of course it's an "Update" :) Let's have a more useful name please, like "German translation" - how hard is that really? (ok, I will do this myself: done!).
> 
> Oh and don't get so defensive that you start to defy rationality! You've asked me to consider your PR for merging so I'll review it :). Do not be surprised if ever some of your changes are not accepted as is as maintainers will often be opinionated and that's ok - it's their rodeo. It's also an opportunity to learn (for both of us).
> 
> On the contrary, you should be grateful I looked at this during my weekend!
> 
> Are you trying to win a collaborator here or annoy someone? :)

I am trying to justify my commits, yes! That's why I forbid myself to be accused of things! And one don't use strings in translation files for configurations, that's what one have configuration files or the database for! No one uses translation strings for this...

**Comment by Justman100** [comment]
But please...

**Comment by merefield** [comment]
> No one uses translation strings for this...

What do you mean?

This text is ultimately output to the screen, so we are using a translation file to determine what the user ends up seeing.

This is ultimately no different to this:

https://github.com/discourse/discourse/pull/26109/files

Note how they've changed the default here and it's nothing to do with a translation.

Are you saying Discourse team don't know what they are doing?

Are you saying that we should have a config setting that determines if the button should say "Send Message" or "Message"?!

In the same way we want to give the admin some control over how discord usernames are presented, so we provide a default "discord_... " , e.g. "discord_bob", but make it easy for them to change it to whatever, e.g. "disco_server_bob" ... or etc.  We absolutely should and would put that in a translation file.

It's called "Text Customisation" and is a completely valid use of a locale file and it is why Discourse has the Admin -> Customize -> Text menu!

**Comment by merefield** [comment]
thanks!

**Comment by Justman100** [comment]
> > No one uses translation strings for this...
> 
> What do you mean?
> 
> This text is ultimately output to the screen, so we are using a translation file to determine what the user ends up seeing.
> 
> This is ultimately no different to this:
> 
> https://github.com/discourse/discourse/pull/26109/files
> 
> Note how they've changed the default here and it's nothing to do with a translation.
> 
> Are you saying Discourse team don't know what they are doing?
> 
> Are you saying that we should have a config setting that determines if the button should say "Send Message" or "Message"?!
> 
> In the same way we want to give the admin some control over how discord usernames are presented, so we provide a default "discord_... " , e.g. "discord_bob", but make it easy for them to change it to whatever, e.g. "disco_server_bob" ... or etc. We absolutely should and would put that in a translation file.
> 
> It's called "Text Customisation" and is a completely valid use of a locale file and it is why Discourse has the Admin -> Customize -> Text menu!

However, this is about variables and placeholders, which are not handled as configuration, not as string

**Comment by merefield** [comment]
> However, this is about variables and placeholders, which are not handled as configuration, not as string

We are going to have to agree to disagree here.  I'm not convinced by your argument.

Thanks very much for the PR in any case!

**Pull request merged**